service/storage_proxy.cc

```diff
@@ -726,12 +726,35 @@ class storage_proxy::remote {
         auto cmd = make_lw_shared<query::read_command>(std::move(cmd1));
         auto src_ip = src_addr.addr;
         auto timeout = t ? *t : db::no_timeout;
+
+        // Pull a schema from the coordinator. There are two cases:
+        // 1. A mixed nodes cluster, cmd->schema_version is a table schema, a table schema is pulled
+        // 2. All new nodes cluster, cmd->schema_version is a reversed table schema, a reversed table schema is pulled.
+        //    Note that in this case, table schema is also pulled if outdated or missing.
         auto f_s = co_await coroutine::as_future(get_schema_for_read(cmd->schema_version, std::move(src_addr), timeout));
         if (f_s.failed()) {
             co_return co_await encode_replica_exception_for_rpc<Result>(p->features(), f_s.get_exception());
         }
         schema_ptr s = f_s.get();
 
+        // Detect whether a transformation from legacy reverse format into native reverse
+        // format is necessary before executing the read_command. That happens when the
+        // native_reverse_queries feature is turned off. A custer has mixed nodes.
+        bool format_reverse_required = false;
+
+        // Check if we have a reversed query
+        if (cmd->slice.is_reversed()) {
+            // Verify whether read_command is provided in legacy or native reversed format by comparing
+            // the schema version provided by read_command with the table schema replica holds. Note that
+            // with the get_schema_for_read call above, table_schema is up-to-date.
+            auto table_schema = p->local_db().find_schema(cmd->cf_id);
+            // Therefore there are only two options:
+            // 1. versions are the same -> legacy format
+            // 2. versions are different -> schema version is equal to reversed table schema version -> native format
+            format_reverse_required = cmd->schema_version == table_schema->version() ? true : false;
```

Comment:
Why not just directly assign:
```c++
format_reverse_required = cmd->schema_version == table_schema->version();
```
This is perfectly valid C++. The ?: operator makes this more confusing.

Reply:
That's so true. I was just so focused on removing the previous function call that produced this.